sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsAsyncClient.java

```diff
@@ -92,67 +94,282 @@ public HttpPipeline getHttpPipeline() {
+     * @return The application/json relationship created.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<String> createRelationship(String digitalTwinId, String relationshipId, String relationship) {
+        return withContext(context -> createRelationship(digitalTwinId, relationshipId, relationship, context))
+            .map(DigitalTwinsResponse::getValue);
+    }
+
+    /**
+     * Creates a relationship on a digital twin.
+     *
+     * @param digitalTwinId The Id of the source digital twin.
+     * @param relationshipId The Id of the relationship to be created.
+     * @param relationship The application/json relationship to be created.
+     * @return A REST response containing the application/json relationship created.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<DigitalTwinsResponse<String>> createRelationshipWithResponse(String digitalTwinId, String relationshipId, String relationship) {
+        return withContext(context -> createRelationship(digitalTwinId, relationshipId, relationship, context));
+    }
+
+    Mono<DigitalTwinsResponse<String>> createRelationship(String digitalTwinId, String relationshipId, String relationship, Context context) {
         return protocolLayer
             .getDigitalTwins()
-            .addWithResponseAsync(digitalTwinId, digitalTwin, context)
-            .flatMap(
-                response -> {
-                    T genericResponse = mapper.convertValue(response.getValue(), klazz);
+            .addRelationshipWithResponseAsync(digitalTwinId, relationshipId, relationship, context)
+            .flatMap(response -> {
+                try {
+                    String jsonResponse = mapper.writeValueAsString(response.getValue());
                     DigitalTwinsResponseHeaders twinHeaders = mapper.convertValue(response.getDeserializedHeaders(), DigitalTwinsResponseHeaders.class);
-                    return Mono.just(new DigitalTwinsResponse<>(response.getRequest(), response.getStatusCode(), response.getHeaders(), genericResponse, twinHeaders));
-                });
+                    return Mono.just(new DigitalTwinsResponse<>(response.getRequest(), response.getStatusCode(), response.getHeaders(), jsonResponse, twinHeaders));
+                } catch (JsonProcessingException e) {
+                    return Mono.error(e);
+                }
+            });
     }
 
     /**
      * Creates a relationship on a digital twin.
      *
      * @param digitalTwinId The Id of the source digital twin.
      * @param relationshipId The Id of the relationship to be created.
-     * @param relationship The application/json relationship to be created.
-     * @return A REST response containing the application/json relationship created.
+     * @param relationship The relationship to be created.
+     * @param clazz The model class to convert the relationship to.
+     * @param <T> The generic type to convert the relationship to.
```

Comment:
Is this also required?

Reply:
Yes, the javadoc generator reports an error otherwise.